### ISSUE TITLE
Memory leak when registering an IAsyncOperation Complete handler before app termination (C++/WinRT & MFC)

### ISSUE THREAD

**Issue by Optimierungswerfer** [MicrosoftEdge/WebView2Feedback#3633]
**Description**
A memory leak is detected when registering a Completed handler on an IAsyncOperation object returned by an async WebView2 API method just before the application terminates. This happens even if the Completed handler is just an empty lambda that captures nothing. It does not happen when the same IAsyncOperation gets run but has no Completed handler.

> ...\WebView2Ctrl.cpp(109) : atlTraceGeneral - [Info] CWebView2Ctrl::ExecuteScript
...\WebView2Ctrl.cpp(99) : atlTraceGeneral - [Info] CWebView2Ctrl::OnDestroy
The thread 0x2260 has exited with code 0 (0x0).
The thread 0x5fd8 has exited with code 0 (0x0).
The thread 0x34f0 has exited with code 0 (0x0).
The thread 0x451c has exited with code 0 (0x0).
The thread 0x418c has exited with code 0 (0x0).
The thread 0x4064 has exited with code 0 (0x0).
The thread 0x3b94 has exited with code 0 (0x0).
The thread 0x56c8 has exited with code 0 (0x0).
The thread 0x33f8 has exited with code 0 (0x0).
The thread 0x6074 has exited with code 0 (0x0).
Detected memory leaks!
Dumping objects ->
{1894} normal block at 0x00000226010F4B10, 16 bytes long.
 Data: <`               > 60 05 8A CA F6 7F 00 00 CD CD CD CD 01 00 00 00 
Object dump complete.
The program '[13824] WebView2MemoryLeakDemo.exe' has exited with code 0 (0x0).

The offending code parts from the attached demo solution ([WebView2MemoryLeakDemo.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/12026486/WebView2MemoryLeakDemo.zip)) are:
```cpp
void CWebView2MemoryLeakDemoView::OnActivateView(BOOL bActivate, CView* pActivateView, CView* pDeactiveView)
{
  if (bActivate && m_pWebView2Ctrl) // Only react on activation, not deactivation
  {
    m_pWebView2Ctrl->ExecuteScript();
  }
  __super::OnActivateView(bActivate, pActivateView, pDeactiveView);
}
```
and
```cpp
void CWebView2Ctrl::ExecuteScript()
{
  if (!m_pWebViewImpl)
  {
    return;
  }
  TRACE(_T("\n[Info] CWebView2Ctrl::ExecuteScript\n"));

  m_pWebViewImpl->m_webView2.ExecuteScriptAsync(L"").Completed([](auto&&, auto&&) {}); // !!! This completed handler causes a 16 byte memory leak when closing the application using X while it was unfocused !!! Using co_await also causes a leak
  m_pWebViewImpl->m_webView2.ExecuteScriptAsync(L""); // Does not cause a leak
}
```

**Version**
SDK: 1.0.1823.32 (latest)
Runtime: Stable 114.0.1823.67 (latest)
Framework: C++/WinRT API of WebView2 in MFC application
OS: Win10 21H2 19044.3086

**Regression**
No/Unknown.

**Repro Steps**
See the attached solution ([WebView2MemoryLeakDemo.zip](https://github.com/MicrosoftEdge/WebView2Feedback/files/12026486/WebView2MemoryLeakDemo.zip)) which was created by these steps:
1. Create a new "MFC App" (Application Type: Single Document, Project Style: MFC Standard)
2. Change project properties to C++20 and Windows SDK 10.0.20348.0
3. Add Nu packages Microsoft.Windows.CppWinRT and Microsoft.Web.WebView2 in their latest version to the project.
4. Embed a CWebView2Ctrl (see provided sources) in the generated default view.
5. Handle the OnActivateView event / window message in the view and make an ExecuteScriptAsync (or any async) call to WebView2.
6. Register a Completed handler for the IAsyncOperation object returned by the previous step.

To cause the leak:
1. Run the application.
2. Move focus away from the application.
3. Close the unfocused application (either by clicking the X or by closing it via the taskbar menu).
4. This will cause the CWebView2Ctrl to briefly be activated before app termination, thus start the async operation, which causes the leak when there is a completed handler.

**Additional context**
Cancelling the IAsyncOperation in CWebView2Ctrl::~CWebView2Ctrl before app termination does not help and neither does introducing a sleep (tested to be sure that the async operation definitely has finished before calling Close on WebView2 and destruction of CWebView2Ctrl).

**Comment by Optimierungswerfer** [comment]
Edit: Forgot to include the res folder in WebView2MemoryLeakDemo.zip

**Comment by LiangTheDev** [comment]
This should be the same as https://github.com/MicrosoftEdge/WebView2Feedback/issues/3124 and fixed in Edge version >=116.0.1923.0.

**Comment by Optimierungswerfer** [comment]
I just tested it with the latest Edge canary build (117.0.1991.0) and the issue seems to have been resolved. Thanks!
